apps/prairielearn/src/lib/assessment.js

```diff
@@ -369,12 +370,26 @@ module.exports = {
             );
           }
 
-          const succeeded = !!_.get(content, 'grading.feedback.results.succeeded', true);
-          let gradable = !!_.get(content, 'grading.feedback.results.gradable', true);
+          // There are two "succeeded" flags in the grading results. The first
+          // is at the top level and is set by `grader-host`; the second is in
+          // `results` and is set by course code.
+          //
+          // If the top-level flag is false, that means there was a serious
+          // error in the grading process and we should treat the submission
+          // as not gradable. This avoids penalizing students for issues outside
+          // their control.
+          const jobSucceeded = !!content.grading?.feedback?.succeeded;
+
+          const succeeded = !!(content.grading.feedback?.results?.succeeded ?? true);
```

Comment:
I'm wondering if we need to revisit how the `succeeded` field is handled. Looking at some of our graders, they seem to also use it as a flag for something having gone completely wrong. I understand that we introduced the `gradable` flag for this purpose too though.

Reply:
Related: #7165. I think having "succeeded" in multiple places is confusing, I agree with having a review of its use.

Reply:
Yep, this is definitely confusing!

Reply:
I'll avoid taking any action on this at the moment, but yes, let's rethink this in the future.